### PR TITLE
Add an example for VAECF with CiteULike dataset

### PR DESCRIPTION
**Pull request by redhat6**
### Checklist:

- [ ] I have added tests.
- [ ] I have updated the documentation accordingly.
- [x] I have updated `README.md` (if you are adding a new model).
- [x] I have updated `examples/README.md` (if you are adding a new example).
- [ ] I have updated `datasets/README.md` (if you are adding a new dataset).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -140,7 +140,7 @@ The recommender models supported by Cornac are listed below. Why don't you join
 | 2018 | [Collaborative Context Poisson Factorization (C2PF)](cornac/models/c2pf), [paper](https://www.ijcai.org/proceedings/2018/0370.pdf) | N/A | [c2pf_exp.py](examples/c2pf_example.py)
 |      | [Multi-Task Explainable Recommendation (MTER)](cornac/models/mter), [paper](https://arxiv.org/pdf/1806.03568.pdf) | [requirements.txt](cornac/models/mter/requirements.txt) | [mter_exp.py](examples/mter_example.py)
 |      | [Probabilistic Collaborative Representation Learning (PCRL)](cornac/models/pcrl), [paper](http://www.hadylauw.com/publications/uai18.pdf) | [requirements.txt](cornac/models/pcrl/requirements.txt) | [pcrl_exp.py](examples/pcrl_example.py)
-|      | [Variational Autoencoder for Collaborative Filtering (VAECF)](cornac/models/vaecf), [paper](https://arxiv.org/pdf/1802.05814.pdf) | [requirements.txt](cornac/models/vaecf/requirements.txt) |
+|      | [Variational Autoencoder for Collaborative Filtering (VAECF)](cornac/models/vaecf), [paper](https://arxiv.org/pdf/1802.05814.pdf) | [requirements.txt](cornac/models/vaecf/requirements.txt) | [vaecf_example.py](examples/vaecf_example.py)
```
Comment (saghiles) [inline README.md]: Please consider renaming to `vaecf_citeulike` instead.

File: examples/README.md [hunk 2]
```diff
@@ -56,6 +56,8 @@
 
 [sorec_filmtrust.py](sorec_filmtrust.py) - Social Recommendation using PMF (Sorec) with FilmTrust dataset.
 
+[vaecf_example.py](vaecf_example.py) - Variational Autoencoder for Collaborative Filtering (VAECF) with CiteULike dataset.
```
Comment (saghiles) [inline examples/README.md]: Please consider renaming to `vaecf_citeulike` instead.

File: examples/vaecf_example.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+# limitations under the License.
+# ============================================================================
+"""Example for Variational Autoencoder for Collaborative Filtering for Implicit Feedback Datasets (Citeulike)"""
+
+import cornac
+from cornac.data import Reader
+from cornac.datasets import citeulike
+from cornac.eval_methods import RatioSplit
+
+_, item_ids = citeulike.load_text()
```
Comment (saghiles) [inline examples/vaecf_example.py]: Loading item_ids is unnecessary in this case.

File: examples/vaecf_example.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+data = citeulike.load_data(reader=Reader(item_set=item_ids))
```
Comment (saghiles) [inline examples/vaecf_example.py]: Simply use: `data = citeulike.load_data()`.

File: examples/vaecf_example.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+ratio_split = RatioSplit(data=data, test_size=0.2, exclude_unknowns=True,
+                         verbose=True, seed=123, rating_threshold=0.5)
+
+vaecf = cornac.models.vaecf.recom_vaecf.VAECF(name='VAECF', k=10, h=20, n_epochs=100, batch_size=100, learning_rate=0.001, beta=1.0, trainable=True, verbose=False, seed=123, use_gpu=False)
```
Comment (saghiles) [inline examples/vaecf_example.py]: Can be simplified to `vaecf = cornac.models.VAECF(k=10, h=20, n_epochs=100, batch_size=100, learning_rate=0.001, beta=1.0, seed=123)
`

File: README.md (hunk 1 above)
Comment (redhat6) [reply README.md]: OK, I will rename vaecf_example to vaecf_citeulike

File: examples/vaecf_example.py (hunk 3 above)
Comment (redhat6) [reply examples/vaecf_example.py]: Thank you for your review.

File: examples/vaecf_example.py (hunk 5 above)
Comment (redhat6) [reply examples/vaecf_example.py]: Thank you for your review.

File: examples/vaecf_example.py (hunk 3 above)
Comment (saghiles) [reply examples/vaecf_example.py]: Welcome :). Lines 18 (import Reader) and 22 (load_text()) can be removed.

Comment (saghiles) [reply examples/vaecf_example.py]: Almost there :). Please consider removing line 18: `from cornac.data import Reader`.

**Comment by saghiles** [comment]
All looks good. If you don't additional changes, I will merge it. Thanks for contributing!

**Comment by redhat6** [comment]
@saghiles  Thank you for all your help

**Comment by saghiles** [comment]
@redhat6 Welcome!
